### PR TITLE
feat: Added Persistent Process Utils

### PR DESCRIPTION
**Pull request by CoshUS**
#### Which issue(s) does this change fix?
N/A

#### Why is this change necessary?
Adding persistent process testing utilities for using in watch, logs, and traces integration tests.
No prod changes.
#### How does it address the issue?
N/A
#### What side effects does this change have?
N/A
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/testing_utils.py [hunk 1]
```diff
@@ -61,6 +67,100 @@ def run_command_with_inputs(command_list: List[str], inputs: List[str], timeout=
...
+                if callback(output, outputs):
+                    result_queue.put(True)
+                    return
+        except Exception as ex:
+            result_queue.put(ex)
+
+    reading_thread = Thread(target=_read_output, daemon=True)
+    reading_thread.start()
+    reading_thread.join(timeout=timeout)
+    if reading_thread.isAlive():
```
Comment (mndeveci) [inline tests/testing_utils.py:154]: Should this method kill the process if it reaches after timeout?

Comment (CoshUS) [reply tests/testing_utils.py:154]: I want to leave that decision up for each individual test case instead. There might be valid use cases where we want to read another line even if the current line timed out.
For the watch integration test, I put the code for killing the process in teardown instead.

Comment (mndeveci) [inline tests/testing_utils.py:154]:
```suggestion
    if reading_thread.is_alive():
```

**Review by mndeveci — APPROVED** [review]
One final suggestion to fix the issues with py39, other than that it looks good 👍🏼

**Review by mingkun2020 — APPROVED** [review]

**Pull request merged**